src/views/DPI/components/DpiPriceChanges.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { ReactNode } from 'react'
+import styled from 'styled-components'
+import numeral from 'numeral'
+
+import InfoSection from './InfoSection'
+
+import useDpiTokenMarketData from 'hooks/useDpiTokenMarketData'
+
+const DpiPriceChanges: React.FC = () => {
+  const { prices } = useDpiTokenMarketData()
+  const [startTime] = prices?.[0] || [0]
+  const [latestTime, latestPrice] = prices?.[prices.length - 1] || [0, 0]
+  // Coingecko API returns each item as hourly data if < 90 days and daily data > 90 days
+  // So if time between start and end time < 90 days, 1 day of data is 24 array items.
+  const dataCadenceFor24Hours = latestTime - startTime < 7776000000 ? 24 : 1
+  const priceChangeIntervals = [
+    ['1 Day', 1],
+    ['1 Month', 30],
+    ['3 Months', 90],
+    ['1 Year', 365],
+  ]
+
+  const calculatePriceDiffs = (daysOfComparison: number) => {
+    // placeholder if no data available for timerange
+    if (!prices || prices.length < daysOfComparison * dataCadenceFor24Hours)
+      return <div>---</div>
+    const startPriceIndex =
+      prices.length - daysOfComparison * dataCadenceFor24Hours
+    const startPrice = prices[startPriceIndex][1]
+    const priceDiff = ((latestPrice - startPrice) / startPrice) * 100
+    const StyledPrice =
+      priceDiff < 0 ? StyledNegativePrice : StyledPositivePrice
+    return (
+      <StyledPrice>{numeral(priceDiff).format('+0.00a') + '%'}</StyledPrice>
+    )
+  }
+
+  const renderPriceChanges = (config: any) => {
+    const [dateString, daysOfComparison] = config // how to destructure in params without tsx bitching?
+    return (
+      <StyledPriceChange key={daysOfComparison}>
+        <StyledDateString>{dateString}</StyledDateString>
+        {calculatePriceDiffs(daysOfComparison)}
+      </StyledPriceChange>
+    )
```

Comment:
Yeah, it looks like the positive and negative style price could be moved into this section (unless I'm missing something)

Reply:
I think I had that at first but then the logic for checking for the placeholder '---' before doing numeric operation seemed less clean than the early return in the `calculatePriceDiffs` function. I thought about making a single component called `StyledPrice` which took the % diff as a prop and calculated the color in CSS 
 instead of JS but that didn't seem right either. e.g.
```
const priceDiff = calcDiff(days)

const StyledPrice= styled.div'${({priceDiff}) =>
  "color: ${!Number(priceDiff) /* placeholder */ ? 'white'  : priceDiff < 0 ? 'red' : 'green' "}'

<StyledPrice priceDiff={priceDiff}>{priceDiff}</StyledPrice>
```

The way I have it now just felt the easiest to read for handling the three different cases and stylings.

Reply:
What about this?

```
renderPriceChanges = () => {
  const priceDiff = calcDiff(days)

  if (!priceDiff) {
    return <>---<>
  }

  if (priceDiff > 0) {
    return <positive>....</positive>
  }

  return <negative>....</negative>
}
```
If you feel strongly about this, we can leave as is

Reply:
yea we can do that